Source/WebCore/dom/Element.cpp

```diff
@@ -4428,10 +4429,18 @@ bool Element::hasPointerCapture(int32_t pointerId)
 
 #if ENABLE(POINTER_LOCK)
 
-void Element::requestPointerLock()
+JSC::JSValue Element::requestPointerLock(JSC::JSGlobalObject& lexicalGlobalObject, PointerLockOptions&& options)
 {
-    if (document().page())
-        document().page()->pointerLockController().requestPointerLock(this);
+    RefPtr<DeferredPromise> promise;
+    if (document().page()) {
```

Comment:
Done. I used `if (auto* page = document().page())` to match the style used elsewhere in Element.cpp.

Reply:
We should use `RefPtr` rather than `auto*`, since we're generally [moving towards](https://github.com/WebKit/WebKit/pulls?q=is%3Apr+smart+pointer+is%3Aclosed) using smart pointers everywhere. This is a relatively recent effort, so there will be some code that doesn't follow the new paradigm yet.

Reply:
This doesn't seem to currently be possible:

```
/Users/jahoward/git/ext/WebKit/WebKitBuild/Debug/usr/local/include/wtf/RefPtr.h:36:18: error: no member named 'ref' in 'WebCore::Page'
            ptr->ref();
            ~~~  ^
/Users/jahoward/git/ext/WebKit/WebKitBuild/Debug/usr/local/include/wtf/RefPtr.h:63:58: note: in instantiation of member function 'WTF::DefaultRefDerefTraits<WebCore::Page>::refIfNotNull' requested here
    ALWAYS_INLINE RefPtr(T* ptr) : m_ptr(RefDerefTraits::refIfNotNull(ptr)) { }
                                                         ^
In file included from /Users/jahoward/git/ext/WebKit/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource152.cpp:6:
/Users/jahoward/git/ext/WebKit/Source/WebCore/dom/Element.cpp:4435:23: note: in instantiation of member function 'WTF::RefPtr<WebCore::Page>::RefPtr' requested here
    if (RefPtr page = document().page()) {
                      ^
```

It seems `WebCore::Page` can't currently be wrapped in `RefPtr`.

Can you suggest an alternate approach? Or maybe we should just table this for the moment; I feel like it's unrelated to my patch since I'm not changing how or where `document().page()` is being accessed in `Element::requestPointerLock`.

Reply:
Oh! Yeah, it's fine as-is then.

Reply:
Yeah I don't think WebCore::Page is ref counted